src/gordon_gcp/plugins/publisher.py

```diff
@@ -13,44 +13,222 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
 """
-Client module to publish DNS records from an event message. Once an
-event message is done (either successfully published, or met with
-errors along the way), it will placed into the appropriate channel,
-either the ``success_channel`` or ``error_channel`` to be further
-handled by the ``gordon`` core system.
+Client module to publish DNS records from an event message. Once an event
+message is done (either successfully published, or met with errors along the
+way), it will placed into the appropriate channel, either the
+`success_channel`` or ``error_channel`` to be further handled by the ``gordon``
+core system.
 
 .. attention::
-    The publisher client is an internal module for the core gordon
-    logic. No other use cases are expected.
+    The publisher client is an internal module for the core gordon logic. No
+    other use cases are expected.
 """
 
+import asyncio
+import datetime
+import json
+import logging
+import re
+
 import zope.interface
 from gordon import interfaces
 
+from gordon_gcp import exceptions
+from gordon_gcp.plugins import _utils
 
 __all__ = ('GDNSPublisher',)
 
+HOST = 'https://www.googleapis.com'
+BASE_CHANGES_ENDPOINT = ('{host}/dns/{version}/projects/{project}/managedZones/'
+                         '{managedZone}/changes')
+RESOURCE_RECORDS_ENDPOINT = ('{host}/dns/v1/projects/{project}/managedZones/'
+                             '{managedZone}/rrsets')
```

Comment:
Can we integrate the HOST into the URLs from the start?  It seems like overkill to format it in later since it's the same for all calls.

Reply:
I think we were going to throw version in there too?

Reply:
ISTR we eventually decided to leave version alone.

Reply:
I thought we had decided to be consistent with version across both URLs?

Reply:
I'd prefer to leave version as configureable given we might use alpha/beta endpoints.